### PR TITLE
feat(server): Support Resp3

### PR DESCRIPTION
**Pull request by ashotland**
Accept hello 3 command and switch to resp3 response format

On top of the unit tests added in this PR:
- Created a pytest that replays same set of command (using seeder) against redis and df and compares the DBs.
- - Will followup with a PR for that if deemed wanted
- Ran the client integration test coded in https://github.com/stfnmllr/go-resp3/blob/main/client/test/command_test.go
- - All commands which we support (without bugs) passes except for pubsub which I think fails for unrelated reason
---- Modified connection_test.py to switch to resp3 to verify pubsub works can follow up with a PR for that

In any case we can consider flag protecting this and release it as "beta" for some community testing.

### CODE REVIEW

**Comment by romange** [comment]
@ashotland  please rebase your branch

File: src/facade/reply_builder.h [hunk 1]
```diff
@@ -150,8 +162,17 @@ class RedisReplyBuilder : public SinkReplyBuilder {
   static std::string_view StatusToMsg(OpStatus status);
 
  private:
+  enum Resp3Type {
```
Comment (adiholden) [inline src/facade/reply_builder.h:165]: I would rename this to collection type.
You use it in SendStringCollection which gets some a collection type and only if resp3 is enabled it does some changes based on the type but the type of the collection is not relevant if resp3 is enabled or not

File: src/server/server_family.cc [hunk 2]
```diff
@@ -1695,23 +1695,35 @@ void ServerFamily::Hello(CmdArgList args, ConnectionContext* cntx) {
   // checking for the existence of the command to detect if RESP3 is
   // supported or whether authentication can be performed using HELLO
   // will gracefully fallback to RESP2 and using the AUTH command explicitly.
+  bool is_resp3 = false;
```
Comment (adiholden) [inline src/server/server_family.cc:1698]: update the comment above

**Review by adiholden — APPROVED** [review]

**Review by adiholden — APPROVED** [review]

**Comment by tillkruss** [comment]
Does this add support for `CLIENT TRACKING`?

**Comment by romange** [comment]
nope, not yet :)

**Comment by tillkruss** [comment]
Can you ping me when it is, so we can fuzz test it via Relay?

**Comment by romange** [comment]
Sure

**Pull request merged**